lwt/nbd_lwt_unix.ml

```diff
@@ -54,7 +54,20 @@ let io_complete op fd buffer =
   then Lwt.fail End_of_file
   else return ()
 
-let tls_channel_of_fd fd role () =
+let impatient timeout_seconds f buf =
```

Comment:
That would evaluate `(read_tls buf)` and pass the resulting `unit Lwt.t` to `impatient timeout_seconds` (but wouldn't compile because `unit Lwt.t` is not a function)

I suppose we could do something similar though: `impatient` would take a `()` instead of `buf` and would be called with
`read_tls buf = impatient timeout_seconds (fun () -> read_tls buf)`

Reply:
On the other hand, `impatient` is completely generic already: the arg happens to be called `buf` but can be anything. It could already be called in the way I wrote in my previous comment.

Reply:
I was thinking of just passing in the Lwt thread to `impatient` instead of a function argument.